File: 0_migration_2021/pset1/cash/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    // Prompt user for amount owed

    float n;

    do
    {
        n = get_float("Change owed: ");
    }
    while (n < 0.00);


    // convert dollars to cents and set coin counter to 0

    int cents = round(n * 100);

    int coins = 0;


    // substract coins from owed amount

    while (cents > 0)

    {

        if (cents >= 25)
        {
            cents -= 25;
            coins++;
        }


        else if (cents >= 10)
        {
            cents -= 10;
            coins++;
        }


        else if (cents >= 5)
        {
            cents -= 5;
            coins++;
        }


        else if (cents >= 1)
        {
            cents -= 1;
            coins++;
        }

    }

    printf("%i\n", coins);
}

File: sentimental-cash/cash.py
from cs50 import get_float

while True:
    amount = get_float("Amount: ")
    if amount > 0:
        break

# Number of coins
coins = 0

# Divide by 25 cents
quarters = amount * 100 // 25
print(quarters)
# Divide by 10 cents
# Divide by 5 cents
# Divide by 1 cents